File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0925_Long_Pressed_Name.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0925_Long_Pressed_Name {
    public boolean isLongPressedName(String name, String typed) {
        if (typed.length() < name.length()) {
            return false;
        }
        int i = 0, j = 0;
        while (i < name.length() && j < typed.length()) {
            if (name.charAt(i) == typed.charAt(j)) {
                i++;
                j++;
            } else if (j > 0 && typed.charAt(j) == typed.charAt(j - 1)) {
                j++;
            } else {
                return false;
            }
        }
        while (j < typed.length() && j > 0 && typed.charAt(j) == typed.charAt(j - 1)) {
            j++;
        }
        return i == name.length() && j == typed.length();
    }
}

File: algorithms/algorithms-cplusplus/leetcode/Question_0925_Long_Pressed_Name.cc
#include <string>

using namespace std;

class Solution {
 public:
  bool isLongPressedName(string name, string typed) {
    if (typed.size() < name.size()) {
      return false;
    }
    int i = 0, j = 0;
    while (i < name.size() && j < typed.size()) {
      if (name[i] == typed[j]) {
        i++;
        j++;
      } else if (j > 0 && typed[j] == typed[j - 1]) {
        j++;
      } else {
        return false;
      }
    }
    while (j < typed.size() && j > 0 && typed[j] == typed[j - 1]) {
      j++;
    }
    return i == name.size() && j == typed.size();
  }
};
